java/test/org/openqa/selenium/grid/distributor/selector/DefaultSlotSelectorTest.java

```diff
@@ -131,6 +131,36 @@ public void theMostLightlyLoadedNodeIsSelectedFirst() {
     assertThat(lightest.getSlots().stream()).anyMatch(slot -> expected.equals(slot.getId()));
   }
 
+  @Test
+  public void theNodeWhichHasExceededMaxSessionsIsNotSelected() {
+    Capabilities chrome = new ImmutableCapabilities("browserName", "chrome");
+
+    NodeStatus lightLoad =
+      createNode(ImmutableList.of(chrome), 12, 2);
+    NodeStatus mediumLoad =
+      createNode(ImmutableList.of(chrome), 12, 5);
+    NodeStatus maximumLoad =
+      createNode(ImmutableList.of(chrome), 12, 12);
+
+    Set<SlotId> ids = selector.selectSlot(chrome, ImmutableSet.of(maximumLoad, mediumLoad, lightLoad));
+    SlotId expected = ids.iterator().next();
+
+    // The slot should belong to the Node with light load
+    assertThat(lightLoad.getSlots().stream())
+      .anyMatch(slot -> expected.equals(slot.getId()));
+
+    // The node whose current number of sessions is greater than or equal to the max sessions is not included
+    // Hence, the node with the maximum load is skipped
```

Comment:
There's no assertion for verifying this. You may want to check that `nodeIds` does not contain `maximumLoad.getNodeId()`